src/charm.py

```diff
@@ -3,113 +3,97 @@
+            event: The event fired on upgrade charm.
         """
-        # Fetch the new config value
-        log_level = self.model.config["log-level"].lower()
-
-        # Do some validation of the configuration option
-        if log_level in VALID_LOG_LEVELS:
-            # The config is good, so update the configuration of the workload
-            container = self.unit.get_container("httpbin")
-            # Verify that we can connect to the Pebble API in the workload container
-            if container.can_connect():
-                # Push an updated layer with the new config
-                container.add_layer("httpbin", self._pebble_layer, combine=True)
-                container.replan()
-
-                logger.debug("Log level for gunicorn changed to '%s'", log_level)
-                self.unit.status = ops.ActiveStatus()
-            else:
-                # We were unable to connect to the Pebble API, so we defer this event
-                event.defer()
-                self.unit.status = ops.WaitingStatus("waiting for Pebble API")
-        else:
-            # In this case, the config option is bad, so block the charm and notify the operator.
-            self.unit.status = ops.BlockedStatus("invalid log level: '{log_level}'")
-
-    @property
-    def _pebble_layer(self):
-        """Return a dictionary representing a Pebble layer."""
-        return {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {
-                        "GUNICORN_CMD_ARGS": f"--log-level {self.model.config['log-level']}"
-                    },
-                }
-            },
-        }
-
-
-if __name__ == "__main__":  # pragma: nocover
-    ops.main.main(IsCharmsTemplateCharm)
+        self.reconcile(event)
+
+    def reconcile(self, _: ops.EventBase) -> None:
+        """Reconciliation for the jenkins agent charm."""
+        if not self.model.get_relation(AGENT_RELATION):
+            self.model.unit.status = ops.BlockedStatus("Waiting for relation.")
+            return
+
+        if not self.state.agent_relation_credentials:
+            self.model.unit.status = ops.WaitingStatus("Waiting for complete relation data.")
+            logger.info("Waiting for complete relation data.")
+            return
+
+        self.model.unit.status = ops.MaintenanceStatus("Starting agent service.")
+        try:
+            self.jenkins_agent_service.restart()
+        except service.ServiceRestartError as e:
+            logger.debug("Error restarting the agent service %s", e)
+            self.model.unit.status = ops.BlockedStatus("Readiness check failed")
```

Comment:
Is there a `return` missing here? Perhaps this wasn't covered by a test 😉

Reply:
Should this be an errored status? - I think BlockedStatus is used to indicate a user's action is required.

Reply:
I changed the status to Error in this case